packages/component-base/src/overlay-class-mixin.d.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license
+ * Copyright (c) 2021 - 2023 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import type { Constructor } from '@open-wc/dedupe-mixin';
+
+/**
+ * A mixin that forwards CSS class names to the internal overlay element
+ * by setting the `overlayClass` property or `overlay-class` attribute.
+ */
+export declare function OverlayClassMixin<T extends Constructor<HTMLElement>>(
+  base: T,
+): Constructor<OverlayClassMixinClass> & T;
+
+export declare class OverlayClassMixinClass {
+  /**
+   * Space-separated list of CSS class names to be set on the overlay element.
```

Comment:
> Using classList is a convenient alternative to accessing an element's list of classes as a space-delimited string via [element.className](https://developer.mozilla.org/en-US/docs/Web/API/Element/className).
>
> *(from https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)*

```suggestion
   * A space-delimited list of CSS class names to be set on the overlay element.
```

Reply:
Thanks, updated.

Reply:
> The DOMTokenList interface represents a set of **space-separated** tokens. Such a set is returned by [Element.classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) or [HTMLLinkElement.relList](https://developer.mozilla.org/en-US/docs/Web/API/HTMLLinkElement/relList), and many others.

Well, it seems that both are in use. Skip this comment then.